### ISSUE TITLE
devpod ssh does not find commands "in non-standard path"

### ISSUE THREAD

**Issue by behrica** [loft-sh/devpod#777]
Running a command via `devpod ssh` fails if the command is in a non-standard path (even though its directory is in %PATH of user)

```
 devpod ssh  --context default --user vscode try-langchain-opencity --command `clojure --version' 
```
fails while

```
 devpod ssh  --context default --user vscode try-langchain-opencity --command '/home/vscode/.asdf/shims/clojure --version' 
``` 
works.

"clojure" is in the %PATH of the user vscode, as seen by:

```
[carsten@nixos:~]$ devpod ssh try-langchain-opencity
vscode ➜ ~ $ clojure --version
Clojure CLI version 1.11.1.1413
vscode ➜ ~ $ 
```

**What did you expect to happen instead?**  
Both should work in the same way.

**How can we reproduce the bug?** (as minimally and precisely as possible)  
```
Run `devpod ssh  --user vscode --command "xxx"  `  against a host where user vscode has `xxx` in his %PATH (but in a non standard directory)
```
It does work with commands in `/usr/local/bin` for example.

**Local Environment:**  
- DevPod Version: v0.3.7
- Operating System:  linux 
- ARCH of the OS:  i386 

**DevPod Provider:**  
- Local/remote provider: ssh

**Comment by pascalbreuninger** [comment]
Hi @behrica, thanks for opening the issue. How exactly is the `PATH` being set for the vscode user?

**Comment by behrica** [comment]
Not sure.. somewhere in the bootstraping of the shell ? I know that after ssh into it, it's set.

**Comment by behrica** [comment]
The dev container.json is here:
https://github.com/behrica/try-langchain-opencity

**Comment by behrica** [comment]
I changed the title, as it happens "in general" and not only on "devpod ssh", as well on "devpod up".
do this to reproduce:
```
git clone --single-branch --branch 'devpod-#777'  https://github.com/behrica/clojure-datascience-devcontainer.git
cd clojure-datascience-devcontainer/
devpod up .
```

**Comment by behrica** [comment]
I thing that it "should work" in my case as the standdrd says:
https://containers.dev/implementors/json_reference/#formatting-string-vs-array-properties
that the "string format" I use:
https://github.com/behrica/clojure-datascience-devcontainer/blob/b4e70569c45cc5787cf3ed43df9eaec8714e2412/.devcontainer/devcontainer.json#L32
shoudl be executed via the "shell".
Which I interpret as via the "shell of user vscode", which has the 'clojure' command in its 'PATH".
(which I can verify by login into the dev container and execute 'clojure')

**Comment by behrica** [comment]
for information: The 'clojure' command is indeed in "unsual place":
/home/vscode/.asdf/shims/clojure

The PATH of user vscod becomes this:
(co containing indeed '/home/vscode/.asdf/shims')
```
PATH="/usr/local/python/current/bin:/usr/local/py-utils/bin:/usr/local/share/nvm/current/bin:/usr/local/sdkman/bin:/usr/local/sdkman/candidates/java/current/bin:/usr/local/sdkman/candidates/gradle/current/bin:/usr/local/sdkman/candidates/maven/current/bin:/usr/local/sdkman/candidates/ant/current/bin:/usr/lib/jvm/msopenjdk-current/bin:/home/vscode/.asdf/shims:/home/vscode/.asdf/bin:/usr/local/python/current/bin:/usr/local/py-utils/bin:/usr/local/share/nvm/current/bin:/usr/local/sdkman/bin:/usr/local/sdkman/candidates/java/current/bin:/usr/local/sdkman/candidates/gradle/current/bin:/usr/local/sdkman/candidates/maven/current/bin:/usr/local/sdkman/candidates/ant/current/bin:/usr/lib/jvm/msopenjdk-current/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/vscode/.local/bin"
```

**Comment by pascalbreuninger** [comment]
Yeah it should definitely work. I wasn't able to figure out why this happens as of now, we start `bash` as a login shell and `strace` reports we're reading all relevant files, from `/etc/bash.bashrc` over `~/.profile` to `~/.asdf/asdf.sh`. 
Some other users reported they weren't able to run asdf-linked binaries via [ssh as well](https://stackoverflow.com/questions/72779608/asdf-command-not-found-when-run-a-script-remotely-with-ssh)

**Comment by behrica** [comment]
PATH gets set inside 
/etc/bash.bashrc
```
...
. /home/vscode/.asdf/asdf.sh
....
```

**Comment by behrica** [comment]
it does work when doing this:
```
ssh xxx.devpod bash -i clojure
```

so the asdf related scripts do only setup the path correctly in a interactive shell

**Comment by pascalbreuninger** [comment]
According to `man bash`, it should 
``` When bash is invoked as an interactive login shell, or as a non-interactive shell with the --login option, it first reads and executes commands from the file /etc/profile, if that file exists.  After reading that file, it looks for
       ~/.bash_profile, ~/.bash_login, and ~/.profile, in that order, and reads and executes commands from the first one that exists and is readable.  The --noprofile option may be used when the shell is started to inhibit this behavior.
```

And we start bash as the login shell, that's what leaves me puzzled.

**Comment by behrica** [comment]
I digged deep into this...
The ultimate root-cause for things not working is that the docker images used by devcontaier have some magic, which changes the effects that startup scripts have on environment variables, and ultimately can "revert them".
I tried to "add" an environment variable in all potentially possible startup scripts, but "ssh" did not seem that.
Only when doing "ssh bash -i" so forcing the interactive shell.

**Comment by behrica** [comment]
The only way to solve mm problem is to add this to `devcontainer.json`:

```
"remoteEnv": { "PATH": "${containerEnv:PATH}:/home/vscode/.asdf/shims" },
```

As this is standard confirm and works with DevPod, I think we can close this issue..
It is as well a specific issue for asdf an the way it sets the environment variables into `/etc/bash.bashrc`
My `devcontainer.json` uses as well sdkman and this is able to change the path for ssh.

**Comment by pascalbreuninger** [comment]
@behrica alright, thank you for digging deeper into this! I'm sure this will be helpful for others experiencing this problem with devcontainers as well
